contracts/core/MarginRequirements.sol

```diff
@@ -0,0 +1,230 @@
+            _account
+        ];
+
+        // InitialMargin < Collateral
+
+        // Starts with:
+        // notional (USD) * (initial margin/100) < collateral (#tokens) * collateral price (in USD)
+
+        // initial margin is dividing by 100 since it is a %. Then, 100 moves to the other equation side multiplying:
+        // notional (USD) * initial margin < collateral (#tokens) * collateral price * 100
+
+        // Remaining values are added to ensure both sides of the equation are scaled equally given they differ in decimal amounts
+
+        return
+            _notional.mul(initialMarginRequired).mul(10**collateralDecimals).mul(10**ORACLE_DECIMALS) <
+            _collateralAmount.mul(oracle.getPrice(_collateralAsset)).mul(MAX_INITIAL_MARGIN).mul(10**NOTIONAL_DECIMALS);
+    }
+
+    /**
+     * @notice checks if there is enough collateral to withdraw the desired amount
+     * @param _account account address
+     * @param _notional order notional amount (USD value with 6 decimals)
+     * @param _withdrawAmount desired amount to withdraw (with its respective token decimals)
+     * @param _otokenAddress otoken address
+     * @param _vaultID id of the vault
+     * @param _vault vault struct
+     * @return boolean value stating whether there is enough collateral to withdraw
+     */
+    function checkWithdrawCollateral(
+        address _account,
+        uint256 _notional,
+        uint256 _withdrawAmount,
+        address _otokenAddress,
+        uint256 _vaultID,
+        MarginVault.Vault memory _vault
+    ) external view returns (bool) {
+        // retrieve collateral decimals
+        uint256 collateralDecimals = uint256(ERC20Interface(_vault.collateralAssets[0]).decimals());
+
+        // avoids subtraction overflow
+        require(
+            _withdrawAmount.add(maintenanceMargin[_vaultID]) < _vault.collateralAmounts[0],
+            "MarginRequirements: insufficient collateral"
+        );
+
+        //     InitialMargin + MaintenanceMargin < Collateral - WithdrawAmount
+        // (=) InitialMargin < Collateral - WithdrawAmount - MaintenanceMargin
+
+        // Starts with:
+        // notional (USD) * (initial margin/100) < [collateral (#tokens) - withdrawAmount (#tokens) - maintenanceMargin (#tokens)] * collateral price (in USD)
+
+        // initial margin is dividing by 100 since it is a %. Then, 100 moves to the other equation side multiplying:
+        // notional (USD) * initial margin < [collateral (#tokens) - WithdrawAmount (#tokens) - MaintenanceMargin (#tokens)] * collateral price (in USD) * 100
+
+        // Remaining values are added to ensure both sides of the equation are scaled equally given they differ in decimal amounts
+
+        return
+            _notional.mul(_getInitialMargin(_otokenAddress, _account)).mul(10**collateralDecimals).mul(
+                10**ORACLE_DECIMALS
+            ) <
```

Comment:
lets make it less than or equal to

Reply:
https://github.com/ribbon-finance/GammaProtocol-OTC/pull/2/commits/43dc6a1c7b5ae7ca86b7385ffef1f2249e1ab9c0